src/http_server/http_server.rs

```diff
@@ -1,37 +1,67 @@
 use axum::{routing::get, Router};
-
+use log::info;
 use crate::http_server::routes;
 
+#[derive(Clone, Default)]
 pub struct HttpServer {
     host: String,
     port: String,
+    router: Router,
 }
 
-impl HttpServer {
-    pub fn new(host: String, port: String) -> HttpServer {
-        HttpServer { host, port }
-    }
-    pub async fn server_start(self) {
-        let (app, addr) = HttpServer::init_server_configuration(self).await;
+#[derive(Clone, Default)]
+pub struct HttpServerBuilder {
+    host: Option<String>,
+    port: Option<String>,
+    router: Router,
+}
 
-        axum::Server::bind(&addr.parse().unwrap())
-            .serve(app.into_make_service())
-            .await
-            .unwrap();
+impl HttpServerBuilder {
+    pub fn new() -> Self {
+        HttpServerBuilder::default()
     }
 
-    async fn init_server_configuration(self) -> (Router, String) {
-        let app = HttpServer::init_basic_routes();
-        let addr = format!("{}:{}", self.host, self.port);
+    pub fn host(mut self, host: impl Into<String>) -> Self {
+        self.host = Some(host.into());
+        self
+    }
 
-        return (app.clone(), addr);
+    pub fn port(mut self, port: impl Into<String>) -> Self {
+        self.port = Some(port.into());
+        self
     }
 
-    fn init_basic_routes() -> Router {
-        let app = Router::new()
+    pub fn router(mut self) -> Self {
+        self.router = Router::new()
             .route("/", get(routes::hello_world))
             .route("/users", get(routes::get_users))
             .route("/heaviest_files", get(routes::get_heaviest_files));
-        return app;
+        self
+    }
```

Comment:
Adding a method to a builder without any parameters is equivalent as putting it inside the `build` method, i'd recommend moving the code there or add methods to the `HttpServerBuilder` like `add_route()` that would take the same params as the `Router::route` so that we could add route directly from the `HttpServerBuilder` like this:
```rust
pub fn add_route(mut self,  path: &str, method_router: MethodRouter<S, B>) -> Self {
    self.router.route(path, method_router)
    self 
}